plugins/rikki/heroeslounge/controllers/Match.php

```diff
@@ -24,6 +25,25 @@ public function __construct()
         BackendMenu::setContext('Rikki.Heroeslounge', 'manage-matches','manage-matches');
     }
 
+    public static function formExtendFields($form)
+    {
+        $form->addFields([
+            'division@create' => [
+                'label' => 'Division',
+                'span' => 'left',
+                'type' => 'dropdown',
+                'options' => Division::listDivisionsWithLongTitle(),
+                'showSearch' => false,
+                'emptyOption' => '-- No Division --'
+            ],
+            'round@create' => [
+                'label' => 'Round',
+                'span' => 'auto',
+                'type' => 'number'
+            ]
+            ]);
+    }
```

Comment:
Yeah that's how I understood how logic needs to be divided when using `yaml` forms from the OctoberCMS documentation. It made me feel like the best practice is to define most of the logic on the model class instead of the controller, which makes the current implementation look like a bit of a "hack" to me.

Alternatively we could no longer use `yaml` forms, but I think their ease of use and the required rewrite outweighs doing so.

Reply:
I really disliked having a static method in Match that just returns the result of a static method in Division. With this, it convolutes a bit where the fields are defined, but I think that is ok considering the alternative.
However, I think that the Round can be moved back to the .yaml - at least I see no reason why it has to be here.

Side note: Defining fields for a specific view is something that I believe is good to have in the view or the controller, but not in the model